src/wallet/home_cards/deferred_deep_link/parser.rs

```diff
@@ -0,0 +1,286 @@
+use crate::prelude::*;
+use base64::engine::general_purpose::URL_SAFE_NO_PAD;
+use base64::Engine;
+
+#[async_trait::async_trait]
+pub trait DeferredDeepLinkParser: Send + Sync {
+    async fn parse(&self, encoded_value: String) -> Result<HomeCards>;
+}
+
+pub struct Parser {
+    gateway_client: GatewayClient,
+}
+
+impl Parser {
+    pub fn new(gateway_client: GatewayClient) -> Self {
+        Self { gateway_client }
+    }
+}
+
+#[async_trait::async_trait]
+impl DeferredDeepLinkParser for Parser {
+    async fn parse(&self, encoded_value: String) -> Result<HomeCards> {
+        let decoded = self.decode(encoded_value)?;
+        self.transform_onboarding_deep_link_value(decoded).await
+    }
+}
+
+impl Parser {
+    fn decode(
+        &self,
+        encoded_value: impl AsRef<str>,
+    ) -> Result<OnboardingDeepLinkValue> {
+        let decoded_value_json_bytes = URL_SAFE_NO_PAD
```

Comment:
For the errors thrown, let's forward the error message in the CommonError, it would greatly help to know what exactly failed if it will happen.